src/client/api/pythonApi.ts

```diff
@@ -382,7 +382,14 @@ export class InterpreterService implements IInterpreterService {
             .then((api) => {
                 if (!this.eventHandlerAdded) {
                     this.eventHandlerAdded = true;
-                    api.onDidChangeInterpreter(() => this.didChangeInterpreter.fire(), this, this.disposables);
+                    api.onDidChangeInterpreter(
+                        () => {
+                            this.didChangeInterpreter.fire();
+                            this.workspaceCachedActiveInterpreter.clear();
```

Comment:
This restores a single line of code that was dropped in https://github.com/microsoft/vscode-jupyter/pull/5199. The original code is https://github.com/microsoft/vscode-jupyter/commit/1c4fad8c818603309c0ca881cae3c75282d2f881#diff-f851568311c1811b9b71d6bb014982757402f39702bb9df14075e1954199abc1L301-L305